save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -44,11 +42,19 @@ class WarnPlugin(
         testConfig.validateAndSetDefaults()
 
         val warnPluginConfig = testConfig.pluginConfigs.filterIsInstance<WarnPluginConfig>().single()
-        val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().singleOrNull()
+        val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().single()
 
-        return files.chunked(warnPluginConfig.batchSize ?: 1).map { chunk ->
-            handleTestFile(chunk.map { it.single() }, warnPluginConfig, generalConfig)
-        }.flatten()
+        // Special trick to handle cases when tested tool is able to process directories.
+        // In this case instead of executing the tool with file names, we execute the tool with directories.
+        //
+        // In case user wants to use directory mode, he needs simply not to pass [wildCardInDirectoryMode] and it will be null
+        return warnPluginConfig.wildCardInDirectoryMode?.let {
+            handleTestFile(files.map { it.single() }.toList(), warnPluginConfig, generalConfig).asSequence()
+        } ?: run {
```

Comment:
Useless `run`? IIRC diktat should not add this automatically now..

Reply:
Why useless, without run it will be lambda

Reply:
No, I mean, `run` contains a single statement, it can be extracted on its own without any behavior changes